From 4e805ba57d9911e2382c11fb043d6b827de7e369 Mon Sep 17 00:00:00 2001
From: Aleksander Machniak <machniak@kolabsys.com>
Date: Tue, 27 Jun 2017 10:14:27 +0000
Subject: [PATCH 4/4] T2519: Fix Recurrence element structure

For example DayOfWeek=0 is invalid. We make sure no such "empty" values are set.
---
 lib/kolab_sync_data.php | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/lib/kolab_sync_data.php b/lib/kolab_sync_data.php
index b03d04b..c03be25 100644
--- a/lib/kolab_sync_data.php
+++ b/lib/kolab_sync_data.php
@@ -1562,8 +1562,13 @@ abstract class kolab_sync_data implements Syncroton_Data_IData
             break;
         }
 
+        // Skip all empty values (T2519)
+        if ($recurrence['type'] != self::RECUR_TYPE_DAILY) {
+            $recurrence = array_filter($recurrence);
+        }
+
         // required field
-        $recurrence['interval'] = $r['INTERVAL'] ? $r['INTERVAL'] : 1;
+        $recurrence['interval'] = $r['INTERVAL'] ?: 1;
 
         if (!empty($r['UNTIL'])) {
             $recurrence['until'] = self::date_from_kolab($r['UNTIL']);
-- 
2.13.0

